### ISSUE TITLE
New Azerbaijani Translation Progress

### ISSUE THREAD

**Issue by gaearon** [reactjs/az.react.dev#376]
## Maintainer List

* @GasimGasimzada 
* @azizaismayilova 

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the glossary and style guide (once they exist) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find that you can't commit anymore, let the maintainers know so they can assign the page to someone else.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

- [ ] Quick Start (@tesseralis) #12345

When PRs are merged, make sure to mark that page as completed!

## Main Content 

To do before releasing as an "official" translation. **Please translate these pages first.**

Note that each section has an index page, which needs to be translated too.

### Learn React

- [ ] Quick Start
  - [ ] Tutorial: Tic Tac Toe
  - [ ] Thinking in React
  
- [ ] Installation
  - [ ] Start a New React Project
  - [ ] Add React to an Existing Project
  - [ ] Editor Setup
  - [ ] React Developer Tools
  
- [ ] Describing the UI
  - [ ] Your First Component
  - [ ] Importing and Exporting Components
  - [ ] Writing Markup with JSX
  - [ ] JavaScript in JSX with Curly Braces
  - [ ] Passing Props to a Component
  - [ ] Conditional Rendering
  - [ ] Rendering Lists
  - [ ] Keeping Components Pure
  
- [ ] Adding Interactivity
  - [ ] Responding to Events
  - [ ] State: A Component's Memory
  - [ ] Render and Commit
  - [ ] State as a Snapshot
  - [ ] Queueing a Series of State Updates
  - [ ] Updating Objects in State
  - [ ] Updating Arrays in State
  
- [ ] Managing State
  - [ ] Reacting to Input with State
  - [ ] Choosing the State Structure
  - [ ] Sharing State Between Components
  - [ ] Preserving and Resetting State
  - [ ] Extracting State Logic into a Reducer
  - [ ] Passing Data Deeply with Context
  - [ ] Scaling Up with Reducer and Context
  
- [ ] Escape Hatches
  - [ ] Referencing Values with Refs
  - [ ] Manipulating the DOM with Refs
  - [ ] Synchronizing with Effects
  - [ ] You Might Not Need an Effect
  - [ ] Lifecycle of Reactive Effects
  - [ ] Separating Events from Effects
  - [ ] Removing Effect Dependencies
  - [ ] Reusing Logic with Custom Hooks

### API Reference

- [ ] `react`: Hooks
  - [ ] `useCallback`
  - [ ] `useContext`
  - [ ] `useDebugValue`
  - [ ] `useDeferredValue`
  - [ ] `useEffect`
  - [ ] `useId`
  - [ ] `useImperativeHandle`
  - [ ] `useInsertionEffect`
  - [ ] `useLayoutEffect`
  - [ ] `useMemo`
  - [ ] `useReducer`
  - [ ] `useRef`
  - [ ] `useState`
  - [ ] `useSyncExternalStore`
  - [ ] `useTransition`

- [ ] `react`: Components
  - [ ] `<Fragment> (<>)`
  - [ ] `<Profiler>`
  - [ ] `<StrictMode>`
  - [ ] `<Suspense>`

- [ ] `react`: APIs
  - [ ] `createContext`
  - [ ] `forwardRef`
  - [ ] `lazy`
  - [ ] `memo`
  - [ ] `startTransition`

- [ ] `react-dom`: Components
  - [ ] Common (e.g. `<div>`)
  - [ ] `<input>`
  - [ ] `<option>`
  - [ ] `<progress>`
  - [ ] `<select>`
  - [ ] `<textarea>`

- [ ] `react-dom`: APIs
  - [ ] `createPortal`
  - [ ] `flushSync`
  - [ ] `findDOMNode`
  - [ ] `hydrate`
  - [ ] `render`
  - [ ] `unmountComponentAtNode`

- [ ] `react-dom/client`: Client APIs
  - [ ] `createRoot`
  - [ ] `hydrateRoot`

- [ ] `react-dom/server`: Server APIs
  - [ ] `renderToNodeStream`
  - [ ] `renderToPipeableStream`
  - [ ] `renderToReadableStream`
  - [ ] `renderToStaticMarkup`
  - [ ] `renderToStaticNodeStream`
  - [ ] `renderToString`

### Navigation and UI

We suggest to leave *most* of the UI translation until the end. We plan to do some invasive changes to the website folder layout and components, so postponing this until your translation is almost complete would make it easier to merge the changes from our side later. It might make sense to translate the homepage above the fold early, but leave the rest for later. As individual pages get translated, you can change the page titles in the corresponding sidebar files. Finally, when you're translating the navigation, make sure to test both desktop and mobile layouts.

- [ ] Homepage (currently in `HomeContent.js`)
- [ ] Sidebars (currently in `src/sidebar*.json`)
- [ ] Top-level navigation (currently in `TopNav.tsx`)

### When You're Ready...

After everything above is translated, add your language to `deployedLanguages` in `Seo.tsx` of the original [reactjs/react.dev](https://github.com/reactjs/react.dev) repository.

## Secondary Content 

These API pages should ideally be translated too, but they're less urgent and can be done after the others:

- [ ] Legacy React APIs
  - [ ] `Children`
  - [ ] `cloneElement`
  - [ ] `Component`
  - [ ] `createElement`
  - [ ] `createFactory`
  - [ ] `createRef`
  - [ ] `isValidElement`
  - [ ] `PureComponent`

## Optional Content 

These aren't the main translation targets, but if you'd like to do them, feel free to expand the list to include their subpages:

- [ ] Community
- [ ] Blog
- [ ] Warnings

**Comment by nasirmovlamov** [comment]
@gaearon In case of not getting info from them, i would like to participate during translation as a maintainer.

**Comment by gaearon** [comment]
Sounds good — let’s give them some time to respond. Thanks!

**Comment by GasimGasimzada** [comment]
Hi @gaearon, I can spend time on maintaining the repo and helping with translations; however, at a limited capacity in comparison to the first time I did the translations.

**Comment by GasimGasimzada** [comment]
Hi again, @gaearon. I am trying to form a team of contributors to help with the translation. Can you please tell me how to add new people as contributors? I will add them in the repo myself (I think I still have admin rights) but not sure if any other place needs to be updated. @nasirmovlamov and @wamujlb are currently two people who are interested. I am also in contact with others.

**Comment by azizaismayilova** [comment]
Hi @gaearon! I will work on this translation as well. Also, I am happy to review/edit others pr to help them with proofreading.

**Comment by gaearon** [comment]
@GasimGasimzada Please send a PR to edit https://github.com/reactjs/translations.react.dev/blob/main/langs/az.json. I will fix it so that it gets synced later today.

**Comment by vugartaghiyev** [comment]
Hi, I also want to help with the translation.

**Comment by nasirmovlamov** [comment]
I have taken Homepage, i made pr , but i will fix based on our previous dictionary and commit again with fixes

**Comment by nasirmovlamov** [comment]
I have taken `Sidebars`

**Comment by azizaismayilova** [comment]
Hi @nasirmovlamov ! Could you please start translating content in recommended order? As it is mentioned above Navigation and UI content will need some invasive changes.

**Comment by azizaismayilova** [comment]
Hi @vugartaghiyev ! Can you please select the content that you want to work one (preferably in recommended section) and comment it here? İ will add your name to the selected content until @GasimGasimzada will add all of you to the maintainer list.

**Comment by vugartaghiyev** [comment]
Hi @azizaismayilova, I will take `Quick Start` Is that ok?

**Comment by azizaismayilova** [comment]
@vugartaghiyev Yes, sure! I will add your name then.

**Comment by GasimGasimzada** [comment]
@vugartaghiyev @nasirmovlamov @wamujlb please join the Discord to discuss some specific translation topics: https://discord.gg/9ugVdPuf

**Comment by gaearon** [comment]
> Can you please tell me how to add new people as contributors? I will add them in the repo myself (I think I still have admin rights) but not sure if any other place needs to be updated.

I’ve changed how this works yesterday — basically just add people in Settings as admins to the repo.

**Comment by GasimGasimzada** [comment]
awesome will do thank!

**Comment by nasirmovlamov** [comment]
I have taken  `Installation` fyi

**Comment by GasimGasimzada** [comment]
Okay, I added your name near `Installation` and gave you write access.

**Comment by nasirmovlamov** [comment]
Hello, please check new pr and i want to take  `Start a New React Project`

**Comment by GasimGasimzada** [comment]
@nasirmovlamov You can pick it up while we review it in parallel. For the future, when you want to pick up a new page, edit the issue and tag your username next to the page name.

**Comment by nasirmovlamov** [comment]
Sure 👍

**Comment by Vug6r** [comment]
Hey! I would like to take _Describing the UI_

**Comment by nasirmovlamov** [comment]
Sure go ahead @Vug6r

**Comment by exrem** [comment]
hello! `quick start` seems taken but hasn't been translated yet, so, i'd like to take it ^^

**Comment by rarblack** [comment]
I do not see any progress on translation. Is this thread dead?

**Comment by rarblack** [comment]
@gaearon if this thread still valid can you add me to the maintainer list. The progress here is going less than the slow.

**Comment by GasimGasimzada** [comment]
@rarblack added you to maintainers list. Let me know which page you want to pick up.

**Comment by rashadqarayev** [comment]
Hello, i want to join as a contrubuter. I am also author of Javascript book.Can you get in touch with me?
@gaearon
